core/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceType.java

```diff
@@ -19,34 +19,9 @@
 
 package org.elasticsearch.search.aggregations.support;
 
-import org.elasticsearch.common.io.stream.StreamInput;
-import org.elasticsearch.common.io.stream.StreamOutput;
-import org.elasticsearch.common.io.stream.Writeable;
-
-import java.io.IOException;
-
-/*
- * The ordinal values for this class are tested in ValuesSourceTypeTests to
- * ensure that the ordinal for each value does not change and break bwc
- */
-public enum ValuesSourceType implements Writeable<ValuesSourceType> {
-
+public enum ValuesSourceType  {
```

Comment:
When I refactored serialization of aggregations it was a parameter we were serializing but it was inconsistent to do so - every single aggregation subclass knew the value of the parameter. So I made it a required argument to the reading ctor. I kept expecting to hit an aggregation that needed to serialize it but I never did.